src/EFCore.MySql.Json.Microsoft/Infrastructure/IMysqlJsonOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Pomelo Foundation. All rights reserved.
+// Licensed under the MIT. See LICENSE in the project root for license information.
+
+using System.Text.Json;
+
+namespace Pomelo.EntityFrameworkCore.MySql.Json.Microsoft.Infrastructure;
+
+public interface IMysqlJsonOptions
+{
+    JsonSerializerOptions JsonSerializerOptions { get; }
+}
```

Comment:
I am not convinced that this is the right approach. I think users should have the ability to set `JsonSerializerOptions` using the model, not injecting them via a service.

They should be able to set a default on the model itself and then also concrete options on (JSON) properties of entities (that override the default options). They also need to be supplied in a JSON provider specific fashion (since we support Microsoft and Newtonsoft JSON implementations). So this requires JSON provider specific extension methods.